charms/knative-eventing/tests/unit/test_charm.py

```diff
@@ -1,67 +1,64 @@
 # Copyright 2022 Canonical Ltd.
 # See LICENSE file for licensing details.
-#
-# Learn more about testing at: https://juju.is/docs/sdk/testing
 
-import unittest
-from unittest.mock import Mock
+import pytest
+from k8s_resource_handler.exceptions import ErrorWithStatus
+from k8s_resource_handler.lightkube.mocking import FakeApiError
+from ops.model import ActiveStatus, BlockedStatus
 
-from ops.model import ActiveStatus
-from ops.testing import Harness
 
-from charm import OperatorTemplateCharm
+def test_events(harness, mocked_lightkube_client, mocker):
+    # Test install and config_changed event handlers are called
+    harness.begin()
+    install = mocker.patch("charm.KnativeEventingCharm._on_install")
+    config_changed = mocker.patch("charm.KnativeEventingCharm._on_config_changed")
```

Comment:
An alternate way to do these:

```python
    # This replaces the charm instance's _on_install function with the mock directly
    harness.charm._on_install = MagicMock()
    harness.charm.on.install.emit()
    # And you can access it like before, but directly
    harness.charm._on_install.assert_called_once()    
```

Nothing wrong with what you have, but I find for instances of classes this is a bit shorter, works a little better with IDEs (like for renaming things), and less error prone (the string passed to `patch` can get a bit complicated, depending on where the object is coming from)

Reply:
Ah, I didn't knew this. We can use that instead.